### ISSUE TITLE
Everything is an error

### ISSUE THREAD

**Issue by LamprosPitsillos** [pmizio/typescript-tools.nvim#198]
I dont get warnings just errors : 
`*  is declared but its value is never read. [6133]` i expect this to be WARNING but everything is an error.

```lua
require("typescript-tools").setup {
                on_attach = on_attach,
                capabilities = capabilities,
                settings = {
                    separate_diagnostic_server = true,
                    publish_diagnostic_on = "change",
                    expose_as_code_action = {},
                    tsserver_path = nil,
                    tsserver_plugins = {},
                    tsserver_max_memory = "auto",
                    tsserver_format_options = {},
                    tsserver_file_preferences = {},
                    tsserver_locale = "en",
                    complete_function_calls = true,
                    include_completions_with_insert_text = true,
                    code_lens = "off",
                    disable_member_code_lens = true,
                },
            }

```

**Comment by LamprosPitsillos** [comment]
1.
```console
❯ nv -v 
NVIM v0.9.4
Build type: Release
LuaJIT 2.1.1693350652

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "
/nix/store/rf2w155phn9alfh610ldhbyrsnfcczrg-neovim-unwrapped-0.9.4/share/nvim
"

Run :checkhealth for more info
```
2.
```json
  "devDependencies": {
    "@types/node": "^20.9.0",
    "@types/react": "^18.2.15",
    "@types/react-dom": "^18.2.7",
    "@typescript-eslint/eslint-plugin": "^6.0.0",
    "@typescript-eslint/parser": "^6.0.0",
    "@vitejs/plugin-react": "^4.0.3",
    "autoprefixer": "^10.4.16",
    "eslint": "^8.45.0",
    "eslint-plugin-react-hooks": "^4.6.0",
    "eslint-plugin-react-refresh": "^0.4.3",
    "postcss": "^8.4.31",
    "tailwindcss": "^3.3.3",
    "typescript": "^5.0.2",
    "vite": "^4.4.5"
  }

```
3 .
```ts
const a = 13
// ^ a is declared but its value is never read.
```

The issue is that every diagnostic is of type Error even if it isnt like the example above , and makes it difficult to distinguish real error in code.

![swappy-20231207-135224](https://github.com/pmizio/typescript-tools.nvim/assets/61395246/96d5f460-b7fc-4e98-938a-1baba5677e36)

**Comment by pmizio** [comment]
@LamprosPitsillos  For me it is correctly INFO:
![image](https://github.com/pmizio/typescript-tools.nvim/assets/4346598/4907aeab-a8d4-4867-8ec3-5349b7e47ff6)

So there are two things, I had install wrong version of ts because in your package json there is `^` before version. It mean every version greater or equal to in your case `5.02`. Version really depends on entry in your package-lock. But there is simple way to check this run `npx tsc -h` in your project. Help of `tsc` prints version also.
If in this version it still will work for me then it's probably something in your config. We don't have any settings related to error severity. Only way to find what is going on in your case is probably bisect your plugins and config settings.
When we find out w what is probably overriding severity to error then I can try to prevent it on this plugin side.

Long story short please provide me exact typescript version. If I still cannot reproduce then probably I close this issue and please open new one when you trace where is problem.

**Comment by LamprosPitsillos** [comment]
```console
❯ npx tsc --version
Version 5.2.2
```
Also for my entire config if it would help [Conf](https://github.com/LamprosPitsillos/nixos-config/tree/main/home-manager/programs/nvim)

If it still cant be reproduced , could you please tell me the version you used , maybe that will fix it , thanks

**Comment by pmizio** [comment]
ok I'll look into it probably after weekend

**Comment by pmizio** [comment]
@LamprosPitsillos I dived into source code of vscode and I found reason why this happens. They hardcoded some error codes to be translated into warning on editor side :D so I ported this code into plugin and it should be fixed on master now.
Sorry for a little bit of delay.
